scripts/trace_flowcell.py

```diff
@@ -0,0 +1,152 @@
+            action_executions = map(lambda x: x["object_id"], trace_info["action_executions"])
+            for execution in action_executions:
+                yield execution
+
+    def get_actions_from_executions(self, executions):
+        for execution_id in executions:
+            execution_info_response = requests.get(
+                "{}/{}/{}".format(self.api_base_url, "executions", execution_id),
+                verify = self.verify,
+                headers = self.headers)
+            execution_info = json.loads(execution_info_response.text)
+            yield execution_info
+    
+    def filter_and_order_actions_by_name(self, actions):
+        filtered_actions = (action for action in actions if self.workflow in action["action"]["name"])
+        def get_start_time(action):
+            start_time = datetime.strptime(action['start_timestamp'].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+            return start_time
+        sort_actions = sorted(filtered_actions, key = get_start_time)
+        return sort_actions
+
+    def get_sorted_actions_from_tag(self, tag):
+        executions = self.get_executions_for_tag(tag)
+        actions = self.get_actions_from_executions(executions)
+        return self.filter_and_order_actions_by_name(actions)
+
+class PickRunfolder:
+    
+    def __init__(self, config):
+        with open(config, 'r') as ymlfile:
+            cfg = yaml.load(ymlfile)
+        self.hosts = cfg['runfolder_svc_urls']
+
+    def load_folders(self):
+        for host in self.hosts:
+            url_base = '/'.join(host.split('/')[:-1])
+            result = requests.get("{}?state=*".format(url_base))
+            result_json = json.loads(result.text)
+            return result_json["runfolders"]
+
+    def choose_runfolder(self, search_term):
+        all_runfolders = self.load_folders()
+        folders = {}
+        states = {}
+        choice = 0
+        hitcount = 0
+        outstring = "\n\tstatus\trunfolder_link\n"
+        for runfolder in all_runfolders:
+            link = runfolder["link"]
+            state = runfolder["state"]
+            if search_term in link and not "rchive" in link and not "biotank" in link:
+                hitcount += 1
+                dirs = link.split('/')
+                folders[hitcount] = dirs[-1]
+                states[hitcount] = state
+                outstring += "{}\t{}\t{}\n".format(hitcount, state, folders[hitcount])
+        
+        if hitcount < 1:
+            raise ValueError("No hit")
+        if hitcount > 1:
+            print outstring
+            choice = int(raw_input("Which runfolder to trace: "))
+        else:
+            choice = 1
+        return folders[choice], states[choice]
+
+def get_executions(actions):
+    for a in actions:
+        bashCommand = "st2 execution get {}".format(a["id"])
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE) 
+        return process.communicate()
```

Comment:
this method will only lookup the first action and then return, is this the intention?

if yes, I would prefer to expose that logic by just accepting a single action
if no, you could e.g. return a generator (i.e. use `yield` instead of `return`) and iterate over the result

Reply:
good point, will print executions directly